flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/SqlITCase.scala

```diff
@@ -317,4 +320,119 @@ class SqlITCase extends StreamingWithStateTestBase {
     result.addSink(new StreamITCase.StringSink)
     env.execute()
   }
+
+  /** test sliding event-time unbounded window with partition by **/
+  @Test
+  def testUnboundedEventTimeRowWindowWithPartition(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
+    env.setStateBackend(getStateBackend)
+    StreamITCase.testResults = mutable.MutableList()
+    env.setParallelism(1)
+
+    val sqlQuery = "SELECT a, b, c, " +
+      "SUM(a) over (" +
+      "partition by a order by rowtime() range between unbounded preceding and current row), " +
+      "count(a) over (" +
+      "partition by a order by rowtime() range between unbounded preceding and current row), " +
+      "avg(a) over (" +
```

Comment:
Also, most groups have just a single record. The max is two records. With that we cannot really check if the sorting works correctly. Can you make less groups (less distinct `a` values) and add more rows for some groups with out-of-order timestamps?